diff --git a/mcp.cfg b/mcp.cfg
index ba606a7..a135215 100644
--- a/mcp.cfg
+++ b/mcp.cfg
@@ -79,7 +79,7 @@ FFPatchServer = %(DirConf)s/patches/minecraft_server_ff.patch
 [RECOMPILE]
 BinClient       = %(DirBin)s/minecraft
 BinServer       = %(DirBin)s/minecraft_server
-ClassPathClient = %(DirLib)s/,%(DirLib)s/*,%(DirJars)s/bin/minecraft.jar,%(DirJars)s/bin/jinput.jar,%(DirJars)s/bin/lwjgl.jar,%(DirJars)s/bin/lwjgl_util.jar
+ClassPathClient = %(DirLib)s/,%(DirLib)s/*,%(DirJars)s/bin/minecraft.jar,%(DirJars)s/bin/jinput.jar,%(DirJars)s/bin/lwjgl.jar,%(DirJars)s/bin/lwjgl_util.jar,%(DirJars)s/bin/AMEEJavaClient.1.6.jar,%(DirJars)s/bin/commons-codec.jar,%(DirJars)s/bin/commons-logging.jar,%(DirJars)s/bin/commons-httpclient.jar,%(DirJars)s/bin/joda-time-1.6.jar,%(DirJars)s/bin/org.json.jar
 ClassPathServer = %(DirLib)s/,%(DirLib)s/*,%(DirJars)s/minecraft_server.jar
 ClientFixes     = %(DirConf)s/patches
 
