diff --git a/minecraft/net/minecraft/src/WorldProvider.java b/minecraft/net/minecraft/src/WorldProvider.java
index b844e4a..3dde35e 100644
--- a/minecraft/net/minecraft/src/WorldProvider.java
+++ b/minecraft/net/minecraft/src/WorldProvider.java
@@ -1,4 +1,5 @@
 package net.minecraft.src;
+import com.amee.minecraft.Atmosphere;
 
 public abstract class WorldProvider {
    public World worldObj;
@@ -105,6 +106,10 @@ public abstract class WorldProvider {
       var4 *= var3 * 0.94F + 0.06F;
       var5 *= var3 * 0.94F + 0.06F;
       var6 *= var3 * 0.91F + 0.09F;
+      float fogColourScale = Atmosphere.fogColourScale();
+      var4 *= fogColourScale;
+      var5 *= fogColourScale;
+      var6 *= fogColourScale;
       return this.worldObj.getWorldVec3Pool().getVecFromPool((double)var4, (double)var5, (double)var6);
    }
 
@@ -117,7 +122,7 @@ public abstract class WorldProvider {
    }
 
    public float getCloudHeight() {
-      return 128.0F;
+     return 128.0F + Atmosphere.cloudHeightOffset();
    }
 
    public boolean isSkyColored() {
