diff --git a/minecraft/net/minecraft/src/GuiIngame.java b/minecraft/net/minecraft/src/GuiIngame.java
index 826661d..805da9e 100644
--- a/minecraft/net/minecraft/src/GuiIngame.java
+++ b/minecraft/net/minecraft/src/GuiIngame.java
@@ -10,6 +10,8 @@ import java.util.Random;
 import net.minecraft.client.Minecraft;
 import org.lwjgl.opengl.GL11;
 
+import com.amee.minecraft.CarbonCounter;
+
 // Referenced classes of package net.minecraft.src:
 //            Gui, ScaledResolution, EntityRenderer, EntityPlayerSP, 
 //            InventoryPlayer, GameSettings, ItemStack, Block, 
@@ -316,6 +318,9 @@ public class GuiIngame extends Gui
             drawString(fontrenderer, (new StringBuilder()).append("f: ").append(MathHelper.floor_double((double)((mc.thePlayer.rotationYaw * 4F) / 360F) + 0.5D) & 3).toString(), 2, 88, 0xe0e0e0);
             drawString(fontrenderer, (new StringBuilder()).append("Seed: ").append(mc.theWorld.getWorldSeed()).toString(), 2, 104, 0xe0e0e0);
             GL11.glPopMatrix();
+        } else
+        {
+            fontrenderer.drawStringWithShadow("CO2: " + com.amee.minecraft.Atmosphere.totalAsString(), 2, 2, 0xffffff);
         }
         if(recordPlayingUpFor > 0)
         {
