diff --git a/minecraft/net/minecraft/src/WorldGenBigTree.java b/minecraft/net/minecraft/src/WorldGenBigTree.java
index 32d710b..ef39b3d 100644
--- a/minecraft/net/minecraft/src/WorldGenBigTree.java
+++ b/minecraft/net/minecraft/src/WorldGenBigTree.java
@@ -6,6 +6,8 @@ package net.minecraft.src;
 
 import java.util.Random;
 
+import com.amee.minecraft.CarbonCounter;
+
 // Referenced classes of package net.minecraft.src:
 //            WorldGenerator, MathHelper, World
 
@@ -207,6 +209,8 @@ public class WorldGenBigTree extends WorldGenerator
         for(int i1 = j + leafDistanceLimit; l < i1; l++)
         {
             float f = func_526_b(l - j);
+        	if (field_41061_a)
+        		CarbonCounter.absorbIntoBlock(Block.leaves.blockID);
             func_523_a(i, l, k, f, (byte)1, 18);
         }
 
@@ -253,6 +257,8 @@ public class WorldGenBigTree extends WorldGenerator
             ai3[j] = MathHelper.floor_double((double)(ai[j] + k) + 0.5D);
             ai3[byte1] = MathHelper.floor_double((double)ai[byte1] + (double)k * d + 0.5D);
             ai3[byte2] = MathHelper.floor_double((double)ai[byte2] + (double)k * d1 + 0.5D);
+        	if (field_41061_a)
+        		CarbonCounter.absorbIntoBlock(Block.wood.blockID);
             func_41060_a(worldObj, ai3[0], ai3[1], ai3[2], i, 0);
         }
